copilot-widget/lib/contexts/Controller.tsx

```diff
@@ -20,75 +22,93 @@ interface ChatContextProps {
 }
 
 const ChatContext = createContext<ChatContextProps | undefined>(undefined);
+const socket = io('http://localhost:8888');
 
 const ChatProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
-  const { data: idata } = useInitialData()
+  const { data: idata } = useInitialData();
   const [messages, setMessages] = useState<Message[]>([]);
+  const sfx = useSoundEffectes();
+  const [loading, setLoading] = useState(false);
+  const [failedMessage, setError] = useState<FailedMessage | null>(null);
+  const config = useConfigData();
+
+  const {sessionId} = useSessionId(config?.token!)
+
+  socket.connect().on("error", (err) => {
+    console.log("Error", err)
+  })
 
   useLayoutEffect(() => {
     if (idata?.history) {
       setMessages(historyToMessages(idata?.history));
     }
   }, [idata?.history]);
+  
+
+  useEffect(() => {
+    socket.on(`${sessionId}_info`, (msg: string) => {
+      
+    })
+  
+    return () => {
+      socket.off(`${sessionId}_info`);
+    };
+  }, [sessionId, setMessages]);
+
+  useEffect(() => {
+    socket.on(sessionId, (msg: string) => {
+      
+    });
+  
+    return () => {
+      socket.off(sessionId);
+    };
+  }, [sessionId]);
```

Comment:
@ah7255703 these two places need to be filled. All informational updates will show up in the
`${sessionId}_info` event

and final answers will be streamed in `sessionId` event.

Info messages have to be cleared at some point in time